IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -0,0 +1,217 @@
+            if (credential)
+            {
+                [self.cacheObjects setObject:credential forKey:key];
+            }
+            else
+            {
+                MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Credential is nil for key %@ while merging storage credentials.", MSID_PII_LOG_MASKABLE(key));
+            }
+        }
+    });
+}
+
+- (void)removeStoredCredentialForKey:(MSIDDefaultCredentialCacheKey *)key
+{
+    dispatch_barrier_async(self.queue, ^{
+        NSString *credentialKey = [self getCredentialKey:key];
+        MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Removing credential for key %@.", MSID_PII_LOG_MASKABLE(credentialKey));
+        [self.cacheObjects removeObjectForKey:credentialKey];
+    });
+}
+
+- (NSArray<MSIDCredentialCacheItem *> *)storedCredentialsForKey:(MSIDDefaultCredentialCacheKey *)key
+{
+    __block NSArray<MSIDCredentialCacheItem *> *credentials =  @[];
+    
+    dispatch_sync(self.queue, ^{
+        if (key.account && key.service)
+        {
+            NSString *credentialKey = [self getCredentialKey:key];
+            MSIDCredentialCacheItem *credential = [self.cacheObjects objectForKey:credentialKey];
+            if (credential)
+            {
+                credentials = @[credential];
+            }
+        }
+        else
+        {
+            NSArray *storedCredentials = [self.cacheObjects allValues];
+            credentials = [storedCredentials filteredArrayUsingPredicate:[self createPredicateForKey:key]];
+        }
+    });
+    
+    return credentials;
+}
+
+/*
+ This api is thread safe only if an immutable object is passed as parameter.
+ */
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json
+                                 error:(NSError * __autoreleasing *)error
+{
+    MSIDMacCredentialStorageItem *instance = [self init];
+    
+    if (instance)
+    {
+        for (NSString *credentialKey in json)
+        {
+            NSDictionary *rtDict = [json msidObjectForKey:credentialKey ofClass:[NSDictionary class]];
+            
+            if (rtDict)
+            {
+                MSIDCredentialCacheItem *credential = [[MSIDCredentialCacheItem alloc] initWithJSONDictionary:rtDict error:error];
+                
+                if (credential)
+                {
+                    [instance.cacheObjects setObject:credential forKey:credentialKey];
+                }
+            }
+        }
+    }
+    
+    return instance;
+}
+
+- (NSDictionary *)jsonDictionary
+{
+    __block NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+    
+    dispatch_sync(self.queue, ^{
+        for (NSString *credentialKey in self.cacheObjects)
+        {
+            MSIDCredentialCacheItem *credential = [self.cacheObjects objectForKey:credentialKey];
+            if (credential)
+            {
+                NSDictionary *atDict = [credential jsonDictionary];
```

Comment:
nit: change name from atDict to just dict? it can be anything, including id token etc...

Reply:
Call it something like 'credentialDict' and add a comment to specify that it stores both AT and IdTokens.